### ISSUE TITLE
Bad Token crash in ConsentForm.show

### ISSUE THREAD

**Issue by djrobinson999** [googleads/googleads-consent-sdk-android#74]
android.view.WindowManager$BadTokenException: 
 at android.view.ViewRootImpl.setView (ViewRootImpl.java:953)
 at android.view.WindowManagerGlobal.addView (WindowManagerGlobal.java:381)
 at android.view.WindowManagerImpl.addView (WindowManagerImpl.java:100)
 at android.app.Dialog.show (Dialog.java:420)
 at com.google.ads.consent.ConsentForm.show (ConsentForm.java:353

This crash happens in 0.7% of my sessions so is quite a big issue.

The problem can happen when the ConsentForm is loaded after the OS has killed the activity that loaded the ConsentForm. Please change your code so that a check is made before Dialog.show that the activity is running.

**Comment by raichy** [comment]
same here!

**Comment by raichy** [comment]
meybe this could help

consentFormListener = new ConsentFormListener() {
            @Override
            public void onConsentFormLoaded() {
                if(!activity.isFinishing()) consentForm.show();
            }

**Comment by djrobinson999** [comment]
Yes I tried that but in the example I managed to reproduce (by killing the top activity using android:noHistory=true) it didn't work. Would rather there was a proper fix to the library before putting out something I can't guarantee will work.

This is my apps main crash by a long way now and its due to this library so it would be good if it could be sorted.

BTW the crash mainly occurs on lower spec phones which would make sense if it's the OS killing the top activity due to resource issues.

**Comment by wurstikon** [comment]
We have this problem too. It's exactly like @djrobinson999 described. The most common crash in our apps and occurs mostly on low end devices. Affects our Android Vitals badly. Waiting for a fix.

**Comment by larykUA** [comment]
have the same problem

**Comment by arammkhitaryan** [comment]
+1

**Comment by rayano24** [comment]
I'm also experiencing this issue, I'd like to know if there is an official fix rather than resorting to try catches
